File: app/src/main/java/com/galtashma/parsedashboard/adapters/ParseAppsAdapter.java
package com.galtashma.parsedashboard.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import androidx.annotation.NonNull;

import com.galtashma.parsedashboard.ParseServerConfig;
import com.galtashma.parsedashboard.R;
import com.lucasurbas.listitemview.ListItemView;

import java.util.List;

/**
 * Created by gal on 3/16/18.
 */

public class ParseAppsAdapter extends ArrayAdapter<ParseServerConfig> {

    public interface ParseAppAdapterListener{
        void onClickOpen(ParseServerConfig config);
        void onClickEdit(ParseServerConfig config);
        void onClickDelete(ParseServerConfig config);
    }

    private ParseAppAdapterListener listener;

    public ParseAppsAdapter(@NonNull Context context, @NonNull List<ParseServerConfig> objects) {
        super(context, R.layout.list_item_card, objects);
    }

    @NonNull
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        final ParseServerConfig server = getItem(position);

        if(convertView == null) {
            convertView = LayoutInflater.from(this.getContext()).inflate(R.layout.list_item_card, parent, false);
        }

        ListItemView item = convertView.findViewById(R.id.parse_server_list_item);
        item.setTitle(server.appName);
        item.setSubtitle(server.serverUrl+"\n"+server.appId);

        item.setOnMenuItemClickListener(clickedItem -> {
            if (clickedItem.getItemId() == R.id.action_edit) {
                notifyEdit(server);
            } else if (clickedItem.getItemId() == R.id.action_remove) {
                notifyDelete(server);
            }
        });
        item.setOnClickListener(view -> notifyClick(server));

        return convertView;
    }

    private void notifyClick(ParseServerConfig config){
        if (listener!=null){
            listener.onClickOpen(config);
        }
    }

    private void notifyEdit(ParseServerConfig config){
        if (listener != null){
            listener.onClickEdit(config);
        }
    }

    private void notifyDelete(ParseServerConfig config){
        if (listener != null){
            this.listener.onClickDelete(config);
        }
    }

    public void setListener(ParseAppAdapterListener listener) {
        this.listener = listener;
    }
}

File: app/src/main/java/com/galtashma/parsedashboard/adapters/ParseAppsAdapter.kt
package com.galtashma.parsedashboard.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.annotation.NonNull
import com.galtashma.parsedashboard.ParseServerConfig
import com.galtashma.parsedashboard.R
import com.lucasurbas.listitemview.ListItemView

/**
 * Created by gal on 3/16/18, rewritten by Cyb3rKo on 05/12/22.
 */

class ParseAppsAdapter(
    @NonNull context: Context,
    @NonNull objects: List<ParseServerConfig>
) : ArrayAdapter<ParseServerConfig>(context, R.layout.list_item_card, objects) {

    interface ParseAppAdapterListener {
        fun onClickOpen(config: ParseServerConfig)
        fun onClickEdit(config: ParseServerConfig)
        fun onClickDelete(config: ParseServerConfig)
    }

    private lateinit var listener: ParseAppAdapterListener

    override fun getView(
        position: Int,
        convertView: View?,
        parent: ViewGroup
    ): View {
        val server = getItem(position)

        var finalView = convertView
        if (convertView == null) {
            finalView = LayoutInflater.from(context)
                .inflate(R.layout.list_item_card, parent, false)
        }

        val item = finalView?.findViewById<ListItemView>(R.id.parse_server_list_item)
        item?.title = server?.appName
        item?.subtitle = "${server?.serverUrl}\n${server?.appId}"

        if (server != null) {
            item?.setOnMenuItemClickListener { clickedItem ->
                if (clickedItem.itemId == R.id.action_edit) {
                    notifyEdit(server)
                } else if (clickedItem.itemId == R.id.action_remove) {
                    notifyDelete(server)
                }
            }
            item?.setOnClickListener { notifyClick(server) }
        }

        return finalView!!
    }

    private fun notifyClick(config: ParseServerConfig) {
        listener.onClickOpen(config)
    }

    private fun notifyEdit(config: ParseServerConfig) {
        listener.onClickEdit(config)
    }

    private fun notifyDelete(config: ParseServerConfig) {
        listener.onClickDelete(config)
    }

    fun setListener(listener: ParseAppAdapterListener) {
        this.listener = listener
    }
}
